server/route.go

```diff
@@ -491,6 +481,20 @@ func routeSid(sub *subscription) string {
 	return fmt.Sprintf("%s%s:%d:%s", qi, RSID, sub.client.cid, sub.sid)
 }
 
+func parseRouteSid(rsid []byte) (cid uint64, sid []byte, ok bool) {
+	if !bytes.HasPrefix(rsid, []byte(QRSID_PREFIX)) {
+		return
+	}
+
+	for i := QRSID_PREFIX_LEN; i < len(rsid); i++ {
+		switch rsid[i] {
+		case ':':
+			return uint64(parseInt64(rsid[QRSID_PREFIX_LEN:i])), rsid[i+1:], true
```

Comment:
I don't understand why the server would crash when receive a corrupted message like `QRSID:1:`, `rsid[i+1:]` is reslicing and never out bound of the array.
I had added some tests for invalid queue sid, the expects as same with use the regex.